src/main/java/com/google/api/codegen/viewmodel/OutputView.java

```diff
@@ -170,33 +172,62 @@ public static Builder newBuilder() {
 
     @AutoValue.Builder
     public abstract static class Builder {
-      public abstract Builder format(String val);
-
-      public abstract Builder args(ImmutableList<PrintArgView> val);
+      public abstract Builder interpolatedString(StringInterpolationView val);
 
       public abstract PrintView build();
     }
   }
 
   @AutoValue
-  abstract class PrintArgView {
+  abstract class WriteFileView implements OutputView {
+    public abstract StringInterpolationView fileName();
 
-    public abstract String formattedName();
+    public abstract VariableView contents();
 
-    @Nullable
-    public abstract TypeModel type();
+    /**
+     * Used in Node.js. Node.js needs to define `const writeFile` or `var writeFile` if used for the
+     * first time. `writerFile` can be reused if defined through `var writeFile` when there are
+     * multiple write-to-file statements.
+     */
+    public abstract boolean isFirst();
+
+    public Kind kind() {
+      return Kind.WRITE_FILE;
+    }
 
     public static Builder newBuilder() {
-      return new AutoValue_OutputView_PrintArgView.Builder();
+      return new AutoValue_OutputView_WriteFileView.Builder();
     }
 
     @AutoValue.Builder
     public abstract static class Builder {
-      public abstract Builder formattedName(String val);
+      public abstract Builder fileName(StringInterpolationView val);
 
-      public abstract Builder type(TypeModel val);
+      public abstract Builder contents(VariableView val);
 
-      public abstract PrintArgView build();
+      public abstract Builder isFirst(boolean val);
+
+      public abstract WriteFileView build();
+    }
+  }
+
+  @AutoValue
+  abstract class StringInterpolationView {
```

Comment:
Maybe a brief comment saying this contains the format string as needed in the language, and the arguments to be used in that string?